### ISSUE TITLE
Feature Request: LDAP Group Membership Resource

### ISSUE THREAD

**Issue by deepbrook** [Ouest-France/terraform-provider-ldap#43]
Ahoi!

We're using your provider to manage LDAP group memberships, but had to jump some hoops to get it running properly:

Due to company policies we do not have full access to LDAP; specifically, we can only manage members of an LDAP group. In order to manage memberships, we had to import the groups into our terraform state, and use the lifecycle block to disable applying any changes when any attribute other than the `members` attribute was changed.

The problem was that we also had to make sure to properly set group names and OU, because otherwise the operation would fail (I'm guessing terraform/the provider uses the values in the config rather than what's in the remote state in this case).

It's working now, but a `ldap_group_membership` resource would've made this significantly easier:

```hcl
resource "ldap_group_membership" "MyUser" {
  group = data.ldap_group.id
  user    = data.ldap_user.id
}
```

I'm not sure how open you guys are to feature requests, but I wanted to throw this out here anyway. 

Thanks!

**Comment by pablo-ruth** [comment]
Hello @deepbrook,

I'm sorry but I think it doesn't fit best practices to create a terraform resource that doesn't match to an actual object in the API (https://www.terraform.io/plugin/hashicorp-provider-design-principles#resource-and-attribute-schema-should-closely-match-the-underlying-api). There is no "membership" object in ldap, it's only an attribute of a group, so coding maintainable logic to allow that can be hard and there will be a conflict issue between the "ldap_group" resource attribute and the "ldap_group_membership" racing for the same attribute. So I'm not very convinced that we should implement this feature.

**Comment by deepbrook** [comment]
Alright, fair enough. Thanks for getting back to me!

**Comment by seanhoughton** [comment]
I was also looking for the membership resources because we need to manage group creation but membership is managed manually which is not possible when the two concepts are tied together. Note that active directory provider written by hashicorp splits the concepts of groups and memberships.

I would suggest removing making the members an output variable of the ldap_group resource and switch to using a distinct membership resource. This also makes the scope of changes much easier to see - your resource change count is proportional to the number of people added and removed from groups rather than just "1 resource changed".

See: https://registry.terraform.io/providers/hashicorp/ad/latest/docs/resources/group_membership

**Comment by pablo-ruth** [comment]
Hi @seanhoughton,
I'm still not convinced that it will fit well in this provider to create a membership resource. But I think I have another way to allow you to create the group via terraform and at the same time manually manage membership.

I created a PR #77 that add the schema option "Computed" in combination with "Optional":
```
Optional:    true,
Computed:    true,
```
If you do not declare the members attribute in your .tf file, Terraform will not manage membership and just read the value from AD. If the "members" attribute is declared then it will still work as before and manage the membership.
```
resource "ldap_group" "group" {
  ou          = "OU=Test,DC=mycompany,DC=local"
  name        = "TestGroup"
  description = "My group description"
}
```

I released the [v0.8.3](https://github.com/Ouest-France/terraform-provider-ldap/releases/tag/v0.8.3) of the provider with this modification. Can you test it and tell me if it resolves your problem?
